### DISCUSSION TITLE
Data cleaning

### DISCUSSION THREAD

**Discussion by rcatlord** [tech-acs/crvsreportpackage#27] [Ideas]
What are the most common data cleaning tasks :broom: that you encounter when handling CRVS data? I have created a [vignette](https://tech-acs.github.io/crvsreportpackage/articles/data_cleaning.html) that runs through a few possible scenarios. However, it would be much more relevant to our NSO colleagues to have some real life examples.

Please can you post some common data cleaning/munging/wrangling problems that you face with CRVS data and how you overcome them below. Thanks!

**Comment by diego-ons** [comment]
Hi

I think one we could experience is to have to convert sex categories, from "F"/"M" or even 1/0 to "male" and "female.

**Comment by jeroenminderman** [comment]
@rcatlord @diego-ons @tbelay @beardl-ons 
Apologies if this is a bit vague (I need to take some more time to look at details), but for potential data cleaning issues/challenges - 

I wonder if it would be worth having something that deals with **ambiguity or inconsistencies in place/region records**.
I suspect there will be multiple different fields/columns that have a record of place, and we may need a way to transparently and repeatably decide which to use as the "core" location one. E.g. what happens if places are entered hierarchically in different fields, and do we need a way to address cases where e.g. "district" has been entered as a hospital name, or town name has been entered as region, etc?

**Comment by tbelay** [comment]
@rcatlord @diego-ons @jeroenminderman @beardl-ons

-  I would also suggest checking for duplicate rows
- Checking for logical error like registration date happening before event date
- Can we also check data type consistency like a date column should be type date (or can be coerced to date without resulting in NA).

**Comment by gulilattesfaye** [comment]
Replacing a missing value with an average would help.

**Comment by rcatlord** [comment]
I have overhauled the [data cleaning vignette](https://tech-acs.github.io/crvsreportpackage/articles/data_cleaning.html) to include all of your suggestions. If you think of anything else just post it here and I'll attempt a solution in the vignette.
